vllm/model_executor/models/utils.py

```diff
@@ -412,6 +412,23 @@ def merge_multimodal_embeddings(
     Merge ``multimodal_embeddings`` into ``inputs_embeds`` by overwriting the
     positions in ``inputs_embeds`` corresponding to placeholder tokens in
     ``input_ids``.
+    
+    ``placeholder_token_id`` can be a list of token ids (e.g, token ids 
```

Comment:
Oh does this mean that `S`, `B` and `E` (as well as `I`) come from the multi-modal embeddings rather than the vocab embeddings in the language model?

Reply:
`S`, `B` and `E`'s embeddings will come from vocab embeddings of the language model, the embeddings for `I` come from the visione encoder - we basically patch the multimodal embeddings with some text embeddings so that the entire thing can be slice-assigned into the text embeddings.

Reply:
Feel free to take a look at the changes in the pixtral.py (line 235 - 250) as an example

Reply:
Got it. Thanks for the explanation.